src/header.jsx

```diff
@@ -94,7 +94,10 @@ const ViewSelector = ({ isGrid, setIsGrid, sortBy, setSortBy }) => {
                               ? _("Display as a list")
                               : _("Display as a grid")}
                           key="view-toggle-action"
-                          onClick={() => setIsGrid(!isGrid)}
+                          onClick={() => setIsGrid(isGrid => {
+                              localStorage.setItem("cockpit-navigator.isGrid", !isGrid ? "true" : "false");
+                              return !isGrid;
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-243-20240306-124050-3aacf570-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)